ProfileManager.js

```diff
@@ -143,49 +143,46 @@ export default class ProfileManager {
       this._cacheContainer.set(capabilityCacheKey, new Cache());
     }
 
-    let promise = this._requests.get(capabilityKey);
-
-    if(promise) {
-      return promise;
+    const capabilityCache = this._cacheContainer.get(capabilityCacheKey);
+    const agentZcap = capabilityCache.get(capabilityKey);
+    if(agentZcap) {
+      return agentZcap;
     }
 
-    promise = (async () => {
-      const capabilityCache = this._cacheContainer.get(capabilityCacheKey);
-      let zcap = capabilityCache.get(capabilityKey);
-      if(!zcap) {
-        const agentSigner = await this.getAgentSigner(
-          {profileAgentId: profileAgent.id, useEphemeralSigner: false});
-        const originalZcap = profileAgent.zcaps[id];
-        if(!originalZcap) {
-          const {id: profileAgentId} = profileAgent;
-          throw new Error(
-            `The agent "${profileAgentId}" does not have the zcap: "${id}"`);
-        }
-        if(useEphemeralSigner) {
-          const localSigner = await this.getAgentSigner(
-            {profileAgentId: profileAgent.id, useEphemeralSigner: true});
-          zcap = await utils.delegateCapability({
-            signer: agentSigner,
-            request: {
-              ...originalZcap,
-              parentCapability: originalZcap,
-              controller: localSigner.id
-            }
-          });
-        } else {
-          zcap = originalZcap;
-        }
-        capabilityCache.set(capabilityKey, zcap);
+    const promise = (async () => {
```

Comment:
Instead of having an inline async function here can we add an internal helper and call that instead?